File: Dev/Frontend/LambdaPP_WebApp_Alpha_3.0/WildTask/src/app/logic/models/Employee_Skill_Bridge.ts
import Employee from "./Employee"
import Skill from "./Skill"

export default class Employee_Skill_Bridge
{
    emp_ID: number
    
    employee: Employee|null
    
    skill_ID: number

    skill: Skill|null

    skill_Level: number
    
    eS_Enabled: Boolean 

    constructor();
    constructor()
    {
        this.emp_ID = 0;
        this.employee = null;
        this.skill_ID = 0;
        this.skill = null;
        this.skill_Level = 0;
        this.eS_Enabled = true;
    }
    
}

File: Dev/WebApp/LambdaPP_ProjectCore_Alpha/LamdbdaPP_WebAPI_Alpha/Database/Models/Employee_Skill_Bridge.cs
using LambdaPP_WebAPI_Alpha.Database.Models;
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace LambdaPP_WebAPI_Alpha.Database.Models
{
    [PrimaryKey(nameof(Emp_ID), nameof(Skill_ID))]
    public class Employee_Skill_Bridge
    {
        [JsonPropertyName("emp_ID")]
        public int Emp_ID { get; set; }
        [JsonPropertyName("employee")]
        public Employee? employee { get; set; }
        [JsonPropertyName("skill_ID")]
        public int Skill_ID { get; set; }
        [JsonPropertyName("skill")]
        public Skill? skill { get; set; }

        [JsonPropertyName("skill_Level")]
        public int Skill_Level { get; set; }
       
        [JsonPropertyName("eS_Enabled")]
        public Boolean ES_Enabled { get; set; }
    }
}
